bezier-rs/lib/src/lib.rs

```diff
@@ -536,6 +538,95 @@ impl Bezier {
 			.filter(|&point| utils::dvec2_approximately_in_range(point, min, max, max_abs_diff).all())
 			.collect::<Vec<DVec2>>()
 	}
+
+	/// Determine if it is possible to scale the given curve, using the following conditions:
+	/// 1. all the control points are located on a single side of the curve.
+	/// See [the offset section](https://pomax.github.io/bezierinfo/#offsetting) of Pomax's bezier curve primer for more details.
+	fn is_scalable(&self) -> bool {
+		// Verify all the control points are located on a single side of the curve.
+		if let BezierHandles::Cubic { handle_start, handle_end } = self.handles {
+			let angle_1 = (self.end - self.start).angle_between(handle_start - self.start);
+			let angle_2 = (self.end - self.start).angle_between(handle_end - self.start);
+			if (angle_1 > 0. && angle_2 < 0.) || (angle_1 < 0. && angle_2 > 0.) {
+				return false;
+			}
+		}
+		// Verify the on-curve point for t = 0.5 occurs roughly in the center of the polygon.
+		let normal_0 = self.normal(0.);
+		let normal_1 = self.normal(1.);
+		let s = normal_0.x * normal_1.x + normal_0.y * normal_1.y;
+		s.acos() < Bezier::SCALABLE_CURVE_MAX_ANGLE
+	}
+
+	/// Split the curve into a number of scalable subcurves. This function may introduce gaps if subsections of the curve are not reducable.
+	/// The function takes the following parameter:
+	/// - `step_size` - Dictates the granularity at which the function searches for reducable subcurves. The default value is `0.01`.
+	///   A small granularity may increase the chance the function does not introduce gaps, but will increase computation time.
+	pub fn reduce_with_step_size(&self, step_size: f64) -> Vec<Bezier> {
```

Comment:
I think it would be better to just have one function, with the last argument an `Option`.

Reply:
Luckily I added only that second function in the most recent commit, so I'll just revert it.